drivers/adc/ad7193/ad7193.c

```diff
@@ -211,34 +223,37 @@ void ad7193_reset(struct ad7193_dev *dev)
 }
 
 /***************************************************************************//**
- * @brief Set device to idle or power-down.
+ * @brief Sets device into the specified operating mode.
  *
- * @param dev      - The device structure.
- * @param pwr_mode - Selects idle mode or power-down mode.
- *                  Example: 0 - power-down
- *                           1 - idle
+ * @param dev        - The device structure.
+ * @param opt_mode	 - Operating mode to be set.
+ * @param modify_cs  - Allows Chip Select to be modified.
  *
  * @return none.
 *******************************************************************************/
-void ad7193_set_power(struct ad7193_dev *dev,
-		      uint8_t pwr_mode)
+void ad7193_set_operating_mode(struct ad7193_dev *dev,
+			       uint8_t opt_mode,
+			       uint8_t modify_cs)
 {
-	uint32_t old_pwr_mode = 0x0;
-	uint32_t new_pwr_mode = 0x0;
-
-	old_pwr_mode  = ad7193_get_register_value(dev,
-			AD7193_REG_MODE,
-			3,
-			1);
-	old_pwr_mode &= ~(AD7193_MODE_SEL(0x7));
-	new_pwr_mode  = old_pwr_mode |
-			AD7193_MODE_SEL((pwr_mode * (AD7193_MODE_IDLE)) |
-					(!pwr_mode * (AD7193_MODE_PWRDN)));
+	uint32_t old_mode = 0x0;
+	uint32_t new_mode = 0x0;
+
+	old_mode  = ad7193_get_register_value(dev,
+					      AD7193_REG_MODE,
```

Comment:
nit: the nr of lines here can be reduced. This applies for all other `ad7193_get_register_value`/`ad7193_set_register_value` in this commit.

Reply:
Sure, will fix them